lib/errors/error-collector.js

```diff
@@ -328,6 +333,17 @@ class ErrorCollector {
     this.seenObjectsByTransaction = Object.create(null)
   }
 
+  _shouldCollectErrors() {
```

Comment:
+2 for this method.  As agents mature they pickup a lot of these "how should I work under these sets of complex configuration" code chunks.  Getting these into named functions and methods that makes their intent clear is a win.